packages/next-adapter/template/next.config.js

```diff
@@ -1,7 +1,9 @@
 // Learn more: https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/guides/using-nextjs.md#withexpo
 
 const { withExpo } = require('@expo/next-adapter');
+const withPlugins = require('next-compose-plugins');
+const withTM = require('next-transpile-modules')(['react-native-web']);
```

Comment:
react-native-web is compiled afaict. https://github.com/necolas/react-native-web/blob/98dcefd9041c10cc95a2385202498b274d926a6b/packages/react-native-web/package.json#L8-L9

Reply:
Yeah I know that it's compiled ahead of time and shouldn't be transpiled -- I have multiple projects where it's working correctly but it looks like the template doesn't compile if we don't transpile `react-native-web`. More context here: https://github.com/expo/expo-cli/issues/3579#issuecomment-882132690

This is weird and to be honest I think I might be missing something. Any idea why this is happening?

Reply:
Have you already tried modifying the RNW babel plugin to use cjs instead? https://github.com/vercel/next.js/blob/e65c56e7e6c16c3c971984ce7acd161bca6b9c2f/examples/with-react-native-web/babel.config.js#L3

Reply:
It doesn't work :(

Reply:
Following up on this, it looks like this error appears only when not transpiling any other modules: https://github.com/expo/expo-cli/issues/3630#issuecomment-888101730. @marklawlor can you confirm that this issue disappeared for you too?

Reply:
@axeldelafosse I've noticed that all reports seem to be on a hello world template with no other dependencies. Once you add another dependency to `next-transpile-modules` it seems you can remove `react-native`.

I'm not sure why, and I haven't looked into it at all. When I updated `next-adapter` none of the logic on the Expo side was changed (just function signatures) - however `next-transpile-modules` was bumped like 3 major versions. The transpilation code inside `webpack-config` has comments that it was adapted from `next-transpile-modules` so I assume there is now some incompatibility edge case.

In terms of effort, I'm not sure if exploring this bug is worth it - especially if the workaround is a simple documentation change that doesn't seem to have any side effects. I personally don't have the current capacity to look into it and would rather put that effort into native ESM support.